### PR TITLE
CURATOR-594: TestingZooKeeperMain isn't setting tickTime, if configured

### PR DESCRIPTION
**Pull request by franz1981**
https://issues.apache.org/jira/browse/CURATOR-594

### CODE REVIEW

**Comment by franz1981** [comment]
The test isn't that elegant, but probably need mockito or some other mocking framework to get it tested, I'm opened to change it, if needed.

File: curator-test/src/test/java/org/apache/curator/test/TestTestingServer.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+      if (defaultZkTickTime > 0) {
+         customTickMs = defaultZkTickTime + (defaultZkTickTime == Integer.MAX_VALUE ? -1 : +1);
+      } else {
+         customTickMs = 100;
+      }
+      final InstanceSpec spec = new InstanceSpec(null, -1, -1, -1, true, -1, customTickMs, -1);
+      final int zkTickTime;
+      try (TestingServer testingServer = new TestingServer(spec, true)) {
+         final File tmpDir = testingServer.getTempDirectory();
+         tmpDir.deleteOnExit();
```
Comment (madrob) [inline curator-test/src/test/java/org/apache/curator/test/TestTestingServer.java:44]: I'm unclear why we need this? Can you add a comment

Comment (franz1981) [reply curator-test/src/test/java/org/apache/curator/test/TestTestingServer.java:44]: I am not familiar with the server files and if there is a I/O crash con close the files won't be deleted

**Comment by franz1981** [comment]
@madrob the CI seems to pass with jdk 11, but not with jdk 8, but cannot say why, let me know if there is anything more I can do to help :+1:

**Comment by eolivelli** [comment]
I have restarted the build on jdk8

**Comment by franz1981** [comment]
@eolivelli I've had a chat with @Randgalt on https://issues.apache.org/jira/browse/CURATOR-595?focusedCommentId=17330211&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-17330211 that's related, because my use case was to use the test server provided by Curator to test if my lease wrapper is correctly behaving, but it seems to fail regularly despite this fix on the tick time on the test provided in the issue I've linked above...
There are other missing behaviours/features of the curator testing server that prevent it to use it for the use case I've posted?
It can help me to understand if makes sense to me to use it, fix it (no idea how) or just use a test container image to get a standalone containerized version to test the curator semaphore against.

File: curator-test/src/test/java/org/apache/curator/test/TestTestingServer.java (hunk 1 above)
Comment (franz1981) [reply curator-test/src/test/java/org/apache/curator/test/TestTestingServer.java:45]: Let me handle this :+1:

**Comment by franz1981** [comment]
Fixed the tmp directory thanks to @eolivelli comment :+1:

**Review by eolivelli — APPROVED** [review]
Lgtm

**Comment by eolivelli** [comment]
Let me merge as soon as Travis passes

**Comment by michaelpearce-gain** [comment]
@eolivelli any update on merging this? its been a month now....

**Comment by eolivelli** [comment]
Sorry. I forgot

**Comment by michaelpearce-gain** [comment]
@eolivelli Thanks

**Pull request merged**